### ISSUE TITLE
StructuredText + Nuxt - Hydration mismatch on new lines `\n`

### ISSUE THREAD

**Issue by Jamiewarb** [datocms/structured-text#27]
## Summary

While using `StructuredText` from `vue-datocms` (v4.0.3) in a Nuxt 3 app (v3.3.2), there can be hydration mismatch warnings. These occur when the following circumstances are met:

- There is serverside rendering, followed by clientside hydration
- The content being rendered by StructuredText contains newline tokens (`\n`), which are produced in the DatoCMS editor by holding shift and pressing enter. `StructuredText` attempts to replace these with `<br>`.

<img width="917" alt="image" src="https://user-images.githubusercontent.com/2754728/227791638-f0719830-f07c-4ba7-a117-b598eb796db0.png">

The reason for the content mismatch is because, on the server, `<p><br></p>` is produced.  
However, during clientside hydration, `<p><br><!-- --></br></p>` is produced instead.

**Serverside render:**
<img width="175" alt="image" src="https://user-images.githubusercontent.com/2754728/227791709-7e4a15d3-eecc-4857-9f74-c1ab09c43ed1.png">

**Clientside hydration:**
<img width="194" alt="image" src="https://user-images.githubusercontent.com/2754728/227791680-1f26c2d1-2527-4aa1-ad72-94d4d6473f91.png">

## Reproduction

https://stackblitz.com/edit/github-fjlqcy?file=app.vue

Observe in console:

<img width="917" alt="image" src="https://user-images.githubusercontent.com/2754728/227791638-f0719830-f07c-4ba7-a117-b598eb796db0.png">

## Debugging

I followed the steps in this article to place a breakpoint in the code after the hydration mismatch is identified and inspect the variables: https://blog.lichter.io/posts/vue-hydration-error/ 

This allows us to see the output before and after hydration

**Comment by Jamiewarb** [comment]
Note that the warning doesn't appear when using `npm run build` and `node .output/server/index.mjs`, but inspecting the serverside output and comparing to the clientside output shows the same behaviour is occurring

**Comment by matjack1** [comment]
Thank you for the report @Jamiewarb we are going to have a look as soon as possible

**Comment by sistrall** [comment]
Hi @Jamiewarb :-)

Thanks again for reporting this problem: I'm going to release a fix in a moment. It's possible to temporary workaround the issue by specifying a custom rendering function for nodes:

```vue
<template>
  <StructuredText :data="data" :renderNode="renderNode" />
</template>

<script setup>
import { StructuredText } from 'vue-datocms';

const renderNode = (tagName, props, childOrChildren) => {
  return h(
    tagName,
    props,
    typeof childOrChildren === 'undefined' || Array.isArray(childOrChildren)
      ? childOrChildren
      : [childOrChildren]
  );
};

const data = {
  value: {
    schema: 'dast',
    document: {
      type: 'root',
      children: [
        {
          type: 'paragraph',
          children: [
            {
              value: 'Test Content\nTest Content',
              type: 'span',
            },
          ],
        },
      ],
    },
  },
  links: [],
  blocks: [],
};
</script>

```

Here is [a working fork](https://stackblitz.com/edit/github-fjlqcy-lditn4?file=app.vue%3AL8) of your example.

**Comment by sistrall** [comment]
@Jamiewarb I just released [version 4.0.4](https://github.com/datocms/vue-datocms/releases/tag/v4.0.4).
